EIPS/eip-2926.md

```diff
@@ -30,57 +30,55 @@ This specification assumes that [EIP-2584](./eip-2584.md) is deployed, and the m
 #### Constants
 
 - `CHUNK_SIZE`: 32 (bytes)
-- `KEY_LENGTH`: 2 (bytes)
-- `MAX_CHUNK_COUNT`: `0xfffc`
-- `VERSION_KEY`: `0xfffd`
-- `CODELEN_KEY`: `0xfffe`
-- `CODEHASH_KEY`: `0xffff`
+- `VERSION_KEY`: `max(u256)`
 - `VERSION`: 0
 - `EMPTY_CODE_ROOT`: `0xc5d2460186f7233c927e7db2dcc703c0e500b653ca82273b7bfad8045d85a470` (==`keccak256('')`)
-- `HF_BLOCK_NUMBER`: to be defined
+- `HF_TIMESTAMP`: to be defined
+- `ITERATOR_STRIDE`: `50_000`
 
 #### Definitions
 
 - `BE(x, N)`: casts `x` to an unsigned integer of `N` bytes and returns its big-endian representation
 
 ### Code merkleization
 
-For an account record `A` with code `C`, the field `A.codeHash` is replaced with `codeRoot`. `codeRoot` is `EMPTY_CODE_ROOT` if `C` is empty. Otherwise it contains the root of `codeTrie`, a [binary trie](https://hackmd.io/uCWOpSrUQaytBgcO0MVkTQ) with the following leaves:
+For an account record `A` with code `C`, two extra optional fields are added: `A.codeSize` and `A.codeRoot`. If `C` is not empty:
 
-- Key: `VERSION_KEY`, value: `BE(VERSION, 1)`
-- Key: `CODELEN_KEY`, value: `BE(length(C), 4)`
-- Key: `CODEHASH_KEY`, value: `keccak256(C)`
-
-In addition to the above, `codeTrie` commits to a list of code `chunks = [(FIO_0, code_0), ..., (FIO_n, code_n)]` which are derived from `C` in a way that:
-
-- `n < MAX_CHUNK_COUNT`.
-- `code_0 || ... || code_n == C`.
-- `length(code_i) == CHUNK_SIZE` where `0 <= i < n`.
-- `length(code_n) <= CHUNK_SIZE`.
-- `FIO_i` is the offset of the first instruction within the chunk. It should only be greater than zero if the last instruction in `code_i-1` is a multi-byte instruction (like `PUSHN`) crossing the chunk boundary. It is set to `CHUNK_SIZE` in the case where all bytes of a chunk are data.
+ - `A.codeSize = BE()`
+ - `A.codeRoot` contains the root of `codeTrie`, a trie with the following leaves:
+   - Key: `VERSION_KEY`, value: `BE(VERSION, 1)`
+   - A list of code `chunks = [(FIO_0, code_0), ..., (FIO_n, code_n)]` which are derived from `C` as follows:
+     - `code_0 || ... || code_n == C`.
+     - `length(code_i) == CHUNK_SIZE` where `0 <= i < n`.
+     - `length(code_n) <= CHUNK_SIZE`.
+     - `FIO_i` is the offset of the first instruction within the chunk. It should only be greater than zero if the last instruction in `code_i-1` is a multi-byte instruction (like `PUSHN`) crossing the chunk boundary. It is set to `CHUNK_SIZE` in the case where all bytes of a chunk are data.
 
 The `i`th element of `chunks` is stored in `codeTrie` with:
 
-- Key: `BE(i, KEY_LENGTH)`
+- Key: `BE(i, 4)`
```

Comment:
I think for MPT implementations the leading zero bytes are stripped of the key... right?